### DISCUSSION TITLE
Approach Guidance

### DISCUSSION THREAD

**Discussion by mattywarr** [elgatosf/elgato#31] [Help]
Hi everyone. I’m new to streamdeck but I’m working with a client who uses it extensively to stream pool matches.

I have a web API, and he has set up buttons in streamdeck to hit that API which does a number of different things. Each button is hooked up to a different command.

However each command button uses both a TableCode and an API Key which is unique to that streamdeck device.

I’d like to be able to distribute some form of package that allows these buttons to be implemented by A N Other clients, but easily allows for those variables to be passed in at the point of import/plugin setup so it is easily usable for their setup.

Any advice on how I can accomplish this in the form of some distributable package?

**Comment by ChekTek** [comment]
Hi @mattywarr 

I'm not sure I fully understand the question, but it seems like you would just want to utilize the [property inspector (UI)](https://docs.elgato.com/streamdeck/sdk/guides/ui) area of Stream Deck to allow users to set the appropriate API key. You'll want to use [Global Settings](https://docs.elgato.com/streamdeck/sdk/guides/settings#global-settings) as these are safe to store API keys and if you [bundle a profile](https://docs.elgato.com/streamdeck/sdk/guides/profiles) it would only take the user updating 1 of these global keys for the entire profile to update. 

https://sdpi-components.dev/docs/getting-started/introduction
